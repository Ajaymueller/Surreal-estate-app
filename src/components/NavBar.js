import React from 'react';
import '../styles/NavBar.css';
import logo from "../logo.png";
import { Link } from 'react-router-dom';



const Navbar = () => {
    return (
        <div className="Nav-bar">
            <div className="logo">
            <img 
        src={logo}
        className="logo" />
            </div>
            <nav>
        <ul className="navbar-links">
            <Link to="/"> <li className="navbar-links-item">View Properties</li></Link>
            <Link to="/add-property" ><li className="navbar-links-item">Add a Property</li></Link>
        </ul>
            </nav>
        </div>
    )
}

export default Navbar;