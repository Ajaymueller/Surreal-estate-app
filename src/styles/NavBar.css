.Nav-bar {
    top: 0;
    border-bottom: 10px solid grey;
    position: relative;
    width: calc(100% - 60px);
    margin: 0 auto;
    margin-bottom: 50px;
    padding: 10px 0;
    background: #333;
    z-index: 1;
    text-align: left;
    padding-right: 2%;
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    flex-direction: column;
    height: 100px;
}

.Navbar nav {
    display: flex;
    flex-direction: row;
    background: #333;
    border-bottom: 10px solid grey;
    margin: 0 auto;
    text-align: left;
    justify-content: flex-start;
    height: 100px;
    width: calc(100% - 60px);
    transition: all .5s ease;
    max-width: 100%;
    min-width: 1000px;
}

.logo {
    height: 100px;
    width: 300px;
    display: inline-block;
}

#nav_links {
    display: block;
    position: relative;
    top: 0;
    width: 80%;
    height: 30px;
}

#nav_links li {
    display: inline-block;
    margin-right: 20px;
    font-size: 25px;
    color:white;
    padding: 7px;
    text-align: center;
    vertical-align: top;
}

#nav_links li:hover {
    color:#BBBBBB;
    border-color:#BBBBBB;
    cursor: pointer;
}

@media only screen and (max-width: 750px) {
    #nav_links li {
        font-size: 20px;
        padding: 0px;
        display: flex;
        flex-wrap: wrap;
    }

    .logo {
        height: 75px;
        width: 200px
    }

    .Navbar nav {
        display: flex;
        flex-wrap: wrap;
        flex-direction: column;
        background: #333;
        margin: 0 auto;
        text-align: left;
        border-bottom: 5px solid grey;
    }
}

@media only screen and (max-width: 450px) {
    #nav_links li {
        font-size: 15px;
        padding: 0px;
        display: flex;
        flex-wrap: wrap;
    }

    .logo {
        width: 175px; 
        height: 50px;
    }

    .Navbar nav {
        height: 75px;
    }
}