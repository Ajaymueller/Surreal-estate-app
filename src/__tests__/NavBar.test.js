import React from 'react';
import { render } from '@testing-library/react';
import NavBar from '../components/NavBar';
import { MemoryRouter } from 'react-router-dom';

const props = {
    onLogin: jest.fn(), 
    userID: 10, 
    onLogout: jest.fn(),
};

describe("NavBar", () => {
    it("component renders with correct props", () => {
        const { asFragment } = render( 
        <MemoryRouter><NavBar {...props} /></MemoryRouter>)
    expect(asFragment).toMatchSnapshot();
    });
    it("displays a logo", () => {
        const { getByAltText } = render( 
        <MemoryRouter><NavBar {...props} /></MemoryRouter>)
        const logo = getByAltText("logo");
        expect(logo).toBeInTheDocument();
    })
    it("displays a list with link to properties and add properties page", () => {
        const { getByText } = render( 
        <MemoryRouter><NavBar {...props} /></MemoryRouter>)
        const viewProperties = getByText("View Properties")
        const addProperty = getByText("Add a Property")
        expect(viewProperties).toBeInTheDocument();
        expect(addProperty).toBeInTheDocument();

    })
});