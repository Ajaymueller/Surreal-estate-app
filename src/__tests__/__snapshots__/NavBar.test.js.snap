// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders with correct props 1`] = `[Function]`;
